tests/backup/backup_upgrade_test.go

```diff
@@ -1546,3 +1547,371 @@ var _ = Describe("{PXBackupClusterUpgradeTest}", Label(TestCaseLabelsMap[PXBacku
+			log.InfoD("Taking backup of application from source cluster before upgrade")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			if preUpgradePxBackupVersion.GreaterThanOrEqual(pxbackupVersion270) {
+				err = os.Setenv("BACKUP_TYPE", "native_csi")
+			} else {
+				err = os.Setenv("BACKUP_TYPE", "azure")
+			}
+			log.FailOnError(err, "Setting BACKUP_TYPE env variable")
+
+			preUpgradeBackupName = fmt.Sprintf("%s-%s", "autogenerated-backup-pre-upgrade", RandomString(4))
+			log.InfoD("creating backup [%s] in source cluster [%s] (%s), organization [%s], in backup location [%s]", preUpgradeBackupName, SourceClusterName, sourceClusterUid, BackupOrgID, backupLocationName)
+			err = CreateBackupWithValidation(ctx, preUpgradeBackupName, SourceClusterName, backupLocationName, backupLocationUID, scheduledAppContexts, labelSelectors, BackupOrgID, sourceClusterUid, "", "", "", "")
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Creation and Validation of backup [%s]", preUpgradeBackupName))
+		})
+
+		Step("Upgrade Px Backup", func() {
+			log.InfoD("Upgrade Px Backup to version %s", LatestPxBackupVersion)
+			err := PxBackupUpgrade(LatestPxBackupVersion)
+			dash.VerifyFatal(err, nil, "Verifying Px Backup upgrade completion")
+		})
+
+		Step("Upgrade the stork version", func() {
+			log.InfoD("Upgrade the stork version")
+			upgradeStorkImageStr = LatestStorkImage
+			log.Infof("Upgrading stork version on source cluster to %s ", upgradeStorkImageStr)
+			err := UpgradeStorkVersion(upgradeStorkImageStr)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verification of stork version upgrade to - %s on source cluster", upgradeStorkImageStr))
+			err = SetDestinationKubeConfig()
+			log.FailOnError(err, "Switching context to destination cluster failed")
+			log.Infof("Upgrading stork version on destination cluster to %s ", upgradeStorkImageStr)
+			err = UpgradeStorkVersion(upgradeStorkImageStr)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verification of stork version upgrade to - %s on destination cluster", upgradeStorkImageStr))
+			err = SetSourceKubeConfig()
+			log.FailOnError(err, "Switching context to source cluster failed")
+		})
+
+		Step("Update the azure cloud account creds to have only the mandatory fields", func() {
+			var credInspectRequest *api.CloudCredentialInspectRequest
+			var credUpdateRequest *api.CloudCredentialUpdateRequest
+			log.InfoD("Update the azure cloud account creds to have only the mandatory fields")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+
+			// Create the inspect request for cloud credential
+			credInspectRequest = &api.CloudCredentialInspectRequest{
+				OrgId:          BackupOrgID,
+				Name:           cloudCredName,
+				IncludeSecrets: true,
+			}
+			cloudCred, err := Inst().Backup.InspectCloudCredential(ctx, credInspectRequest)
```

Comment:
I would be good if we can move this into a function.

Reply:
I thought about that, but it is as good as just calling UpdateCloudCredential inside another helper function.